consensus/bor/heimdallapp/span.go

```diff
@@ -38,23 +37,10 @@ func toSpan(hdSpan *hmTypes.Span) *span.HeimdallSpan {
 }
 
 func toValidatorSet(vs hmTypes.ValidatorSet) valset.ValidatorSet {
-	set := valset.ValidatorSet{
+	return valset.ValidatorSet{
 		Validators: toValidatorsRef(vs.Validators),
 		Proposer:   toValidatorRef(vs.Proposer),
 	}
-
-	// set validator map
-	validatorsMap := make(map[common.Address]int, len(set.Validators))
-
-	for i, val := range set.Validators {
-		validatorsMap[val.Address] = i
-	}
-
-	set.SetMap(validatorsMap)
-
-	set.SetTotalVotingPower(vs.TotalVotingPower())
```

Comment:
Don't we need to fill all fields?

Reply:
Nope, That is calculated in validator-set while processing. In other ways of communications (HTTP/gRPC) also we don't set totalVotingPower and the validator map as they are anyway calculated.